src/java.base/share/classes/jdk/internal/classfile/attribute/StackMapTableAttribute.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.classfile.attribute;
+
+import java.lang.constant.ClassDesc;
+import java.util.List;
+
+import jdk.internal.classfile.Attribute;
+import jdk.internal.classfile.CodeElement;
+import jdk.internal.classfile.Label;
+import jdk.internal.classfile.constantpool.ClassEntry;
+import jdk.internal.classfile.impl.BoundAttribute;
+import jdk.internal.classfile.impl.StackMapDecoder;
+import jdk.internal.classfile.impl.TemporaryConstantPool;
+import jdk.internal.classfile.impl.UnboundAttribute;
+import static jdk.internal.classfile.Classfile.*;
+
+/**
+ * Models the {@code StackMapTable} attribute (JVMS 4.7.4), which can appear
+ * on a {@code Code} attribute.
+ */
+public sealed interface StackMapTableAttribute
+        extends Attribute<StackMapTableAttribute>, CodeElement
+        permits BoundAttribute.BoundStackMapTableAttribute, UnboundAttribute.UnboundStackMapTableAttribute {
+
+    /**
+     * {@return the stack map frames}
+     */
+    List<StackMapFrameInfo> entries();
+
+    public static StackMapTableAttribute of(List<StackMapFrameInfo> entries) {
+        return new UnboundAttribute.UnboundStackMapTableAttribute(entries);
+    }
+
+    /**
+     * The type of a stack value.
+     */
+    sealed interface VerificationTypeInfo {
+        int tag();
+    }
+
+    /**
+     * A simple stack value.
+     */
+    public enum SimpleVerificationTypeInfo implements VerificationTypeInfo {
```

Comment:
I note that in this class we have made the decision to model all the innards (XYZInfo) as nested classes - while in all the other cases XYZInfo are toplevel types. Moving forward, we should pick something consistent.

Reply:
Every case has been considered individually, evaluated on use cases and pros and cons have been weighted. Unified approach across the whole API would be nice, however not so simple and not the highest priority.

Reply:
I've extracted `StackMapFrameInfo` to the top level and moved `VerificationTypeInfo`, however it is still nested.
Let me know if you think we should really avoid all nested or if info inside info is OK.
Thanks.

Reply:
That seems reasonable.